### DISCUSSION TITLE
Why do this all manually? There are far easier ways.

### DISCUSSION THREAD

**Discussion by toydotgame** [samuelmarina/is-even#144] [General]
## When working with Arabic numerals:
Dividing by 2 and getting the remainder (`% 2`) should return `0`, as even numbers have no remainder when divided by 2.
```js
if(int % 2 == 0) {
    return true; // Even.
} else {
    return false; // Odd.
}
```

## When working with written-out numbers:
There could be an internal system to turn strings like `"seven"` to `7`. `"seventy-seven"` could be separated [into `"seventy"` and `"seven"`] and converted to `70` + `7` (thus, `77`). Although it'd be a bit difficult, words like "billion," "million," and "thousand" are not needed in the conversion.
_For example,_ `373,051` is said as "three hundred seventy-three thousand and fifty-one." Only looking at `373` (and removing the "thousand") could be 373 or 373,000, thanks to the word "thousand" denoting it's place value. If `373` is followed by more numbers, for example, then it can be inferred that `373` is _not_ in the hundreds place, but the thousands/millions/billions/etc place.
I don't feel like I explained this right; but I know it _is_ possible to convert from words to numerals.

Once they have been converted to integers, the same `% 2` code [as above] can be run again, and the number's evenness can be returned.

I find it _ridiculously_ ineffective manually writing the evenness of _all_ numbers out, plus numbers are infinite; making this task effectively impossible. (Unless JavaScript has an integer limit, _i.e_ 2^63. I don't know, I use Java)

**Comment by toydotgame** [comment]
**Update:** Converting to numerals from words has been done before. [`npm install words-to-numbers`](https://www.npmjs.com/package/words-to-numbers). (See [finnfiddle/words-to-numbers](https://github.com/finnfiddle/words-to-numbers))

**Comment by dogtopus** [comment]
Mod math is for ~~p\*\*\*\*\*s~~ weaks. Real ~~men~~ 31337 pr0grammers do
```js
return !(int & 1)
```

(EDIT: Use better inclusive languages.)

**Reply by toydotgame** [reply]
~~Real 31337 H4X0R5 use a real language, and not JavaScript.~~
Mod maths is a single operator (`%`), instead of _two_ (`!` and `&`). Also using `!` for a `return` function is pretty cringe. I personally find `mod()` a lot more intuitive than bitwise thingamajigs.

**Comment by Hocuri** [comment]
> plus numbers are infinite

Are you sure? Do you have any source for that?

**Reply by toydotgame** [reply]
Good point. I think I was told by some guy on the street.

**Comment by Hocuri** [comment]
Also, we're almost done, roadmap: https://github.com/samuelmarina/is-even/pull/136/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5. Do you really want to throw away all this work?

**Reply by geekborg** [reply]
> Do you really want to throw away all this work?

This kind of thought-process is called the [sunk cost fallacy effect](https://en.wikipedia.org/wiki/Sunk_cost#Fallacy_effect).
You may want to avoid thinking this way :)

**Reply by toydotgame** [reply]
No-one would look for a library that does this because the first thing StackOverflow and/or Google shows is just to use code like
```js
var isEven = x % 2 == 0;
```
and then to `console.log()` (or do whatever you want with) the value of `isEven`.

This library is no more than a novelty and a fun test to see just how powerful your browser is at rendering GitHub's raw view of the package's files. (I don't think Firefox was designed with 90 MiB pages in mind)

**Reply by Shashank3736** [reply]
> No-one would look for a library that does this because the first thing StackOverflow and/or Google shows is just to use code like
> 
> ```js
> var isEven = x % 2 == 0;
> ```
> 
> and then to `console.log()` (or do whatever you want with) the value of `isEven`.
> 
> This library is no more than a novelty and a fun test to see just how powerful your browser is at rendering GitHub's raw view of the package's files. (I don't think Firefox was designed with 90 MiB pages in mind)

Man This is not a serious project, to tell even or odd. Seriously nobody needs a package to find a number that is even in any language in the world. If you are seriously criticizing the code then you are just being **DUMB**. It's actually a code popularized from [__this video__](https://www.youtube.com/watch?v=Re3rijPsGoY) and if you see the whole video he seriously reviewed an issue which states the same. Do you really think you are the only one who found this thing 😆 ?

**Reply by toydotgame** [reply]
Uh oh looks like we got someone who didn't take it right.
